akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/BasicDirectivesSpec.scala

```diff
@@ -182,5 +182,19 @@ class BasicDirectivesSpec extends RoutingSpec {
         }
       } ~> check { responseAs[String] shouldEqual "1" }
     }
+
+    "return 408 Request Timeout status on timeout" in {
+      val neverEndingEntity = HttpEntity(ContentTypes.`text/plain(UTF-8)`, Source.maybe[ByteString])
+      val timeout = 10.milliseconds
+
+      Post("/abc", neverEndingEntity) ~> {
+        extractStrictEntity(timeout) { _ =>
+          complete("content")
+        }
+      } ~> check {
+        entityAs[String] shouldEqual s"Request timed out after $timeout while waiting for entity data"
```

Comment:
I think I would prefer to make the timeout explicit in this check rather than invoking the `toString` on the `timeout` - but this is OK as well.